json_encoder.go

```diff
@@ -38,14 +38,20 @@ const (
 	_initialBufSize = 1024
 )
 
-var jsonPool = sync.Pool{
-	New: func() interface{} {
-		return &jsonEncoder{
-			// Pre-allocate a reasonably-sized buffer for each encoder.
-			bytes: make([]byte, 0, _initialBufSize),
-		}
-	},
-}
+var (
+	// ErrNilEntry signals that Encoder.WriteEntry was called with a nil *Entry.
+	ErrNilEntry = errors.New("can't encode a nil *Entry")
+	// ErrNilSink signals that Encoder.WriteEntry was called with a nil WriteSyncer.
+	ErrNilSink = errors.New("can't write encoded message a nil WriteSyncer")
+	jsonPool   = sync.Pool{
```

Comment:
nit: blank line to group the errors and the pool separately (i like to related vars together, even using separate blocks for unrelated variables)